src/ReactDOM.rei

```diff
@@ -478,6 +478,51 @@ external unmountComponentAtNode: Dom.element => unit =
 [@mel.module "react-dom"]
 external flushSync: (unit => unit) => unit = "flushSync";
 
+module Experimental: {
+  /*
+     preinitModule and preloadModule options.
+     https://react.dev/reference/react-dom/preinitModule#parameters
+   */
+  [@deriving jsProperties]
+  type preOptions = {
+    /* a required string. It must be 'script'. */
+    [@mel.as "as"] [@mel.optional]
+    _as: option([ | `script]),
+    /*
+        a required string. It must be "anonymous", "use-credentials", and "".
```

Comment:
I think `crossOrigin` could be polyvariant 

```reason
[ 
  | `anonymous 
  |  [@mes.as "use-credentials"] `useCredentials 
  |  [@mes.as ""] `empty
] 
```

I have been try but got some warning
```reason
Alert: unused Unused attribute [@mel.as]
This means such annotation is not annotated properly.
For example, some annotations are only meaningful in externals
```

Maybe you have any suggestions? @davesnx

Reply:
Many of the domProps could be a polyvar, it's probably a nice idea but it doesn't fit in all cases so it becomes a bit tricky. That's why they are mostly strings now, and introducing a polyvar would give the sensation that any polyvar is good idea, which is not really true.

We should not add in this PR the polyvariant.

Reply:
Thanks for your explanations. Yes,  I keep to use `string` on this PR.